compiler-wasm/src/lib.rs

```diff
@@ -1,6 +1,204 @@
+use std::collections::HashMap;
+use std::ffi::OsStr;
+use std::path::Path;
+
+use gleam_core::Error;
 use wasm_bindgen::prelude::*;
 
+use hexpm::version::{Range, Version};
+
+use gleam_core::build::{Mode, Options, Package, ProjectCompiler, Target};
+use gleam_core::config::{Dependencies, Docs, ErlangConfig, PackageConfig, Repository};
+use gleam_core::io::{FileSystemReader, FileSystemWriter};
+use gleam_core::project::{Base16Checksum, ManifestPackage, ManifestPackageSource};
+
+mod wasm_filesystem;
+use wasm_filesystem::WasmFileSystem;
+
+mod log_telemetry;
+use log_telemetry::LogTelemetry;
+
+mod static_files;
+
+const PROJECT_NAME: &str = "gleam-wasm";
+
+/// Compile a set of `source_files` into a different set of source files for the
+/// `target` language.
+///
+/// The `main_source` file is required.
+/// The `additional_source_files` are optional.
+pub fn compile(
+    main_source: &str,
```

Comment:
I think the idea of a main source module makes sense in the playground, but not in the compiler. I think we should include it like a normal module in the source files hashmap.

Reply:
Done :+1: